packages/wonder-blocks-testing/src/fixtures/adapters/adapter.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+import {AdapterGroup, type CloseGroupFn} from "./adapter-group.js";
+import type {
+    Adapter as AdapterInterface,
+    AdapterGroup as AdapterGroupInterface,
+    AdapterGroupOptions,
+} from "../types.js";
+
+/**
+ * Class for implementing a custom adapter.
+ */
+export class Adapter<Options: {...}, Exports: {...}>
+    implements AdapterInterface<Options, Exports>
+{
+    +_name: string;
+    +_closeGroupFn: CloseGroupFn<any, Options, Exports>;
+
+    /**
+     * @param {string} name The name of the adapter.
+     * @param {CloseGroupFn<any, Options, Exports>} closeGroupFn The function
+     * an adapter group should call when the group is closed. This is invoked
+     * by an adapter group when it is closed. This function is where an
+     * adapter implements the logic to generate the actual fixtures for the
+     * adapter's target framework.
+     */
+    constructor(
+        name: string,
+        closeGroupFn: CloseGroupFn<any, Options, Exports>,
+    ) {
+        if (typeof name !== "string") {
+            throw new TypeError("name must be a string");
+        }
+        if (name.trim() === "") {
+            throw new Error("name must be a non-empty string");
+        }
+        if (typeof closeGroupFn !== "function") {
+            throw new TypeError("closeGroupFn must be a function");
+        }
+
+        this._name = name;
+        this._closeGroupFn = closeGroupFn;
+    }
+
+    /**
+     * The name of the adapter.
+     */
+    get name(): string {
+        return this._name;
+    }
```

Comment:
The name keys the options overrides that a group can give. It seemed handly to expose it in case folks want to make sure they are keying those overrides with the right adapter name (like the adapter names in withTestHarness, for example).